cmd/config-reloader/main.go

```diff
@@ -60,6 +66,24 @@ func main() {
 		level.Error(logger).Log("msg", "parsing reloader URL failed", "err", err)
 		os.Exit(1)
 	}
+
+	// Poll ready endpoint indefinitely until it's up and running.
+	req, _ := http.NewRequest(http.MethodGet, *readyURLStr, nil)
+	if err := wait.PollInfinite(2*time.Second, func() (bool, error) {
```

Comment:
As we wait until a condition is met without fancy features offered by wait, and the endpoint this request intends to hit isn't a kubernetes api - can we only use `ticker` like what https://gobyexample.com/tickers describes rather than `wait.PollInfinite`? In this way, we can also not add a dependency from an external kubernetes library.

Let me know what you think.

Reply:
Well after thinking about this again, this comment should be optional to be addressed, as I also think there is no harm to import `wait` here. Using `ticker` would be an optimization, but not required.

If you want to address this comment in this PR, I can take another look after the change. Otherwise, feel free to resolve this comment. Let me know what you think.

Reply:
I think it's a good callout actually.

I went ahead and used a more idiomatic stdlib ticker-based approach. Lemme know what you think!

Reply:
LGTM!